### PR TITLE
Pip cache enabled and shell overrides removed

### PR DESCRIPTION
**Pull request by offa**
- Enable pip cache
- Remove shell override (bash is [default](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsshell) on all non-windows platforms)

### CODE REVIEW

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -24,22 +25,17 @@ jobs:
         pip install -r requirements-dev.txt
     - name: linter (codespell)
       run: pre-commit run --all-files
-      shell: bash
```
Comment (priv-kweihmann) [inline .github/workflows/pythonpackage.yml:27]: Why not be explicit here? I know that on almost all systems bash is the default, but there are some where it isn't, leading to hard to debug issues, when bashisms are used in the called scripts - I would be okay, but then please put some reasoning into the commit message

Comment (offa) [reply .github/workflows/pythonpackage.yml:27]: Thanks for your review.

The reason was to remove unneeded code from each step. However, it's possible to set a global default instead, so the shell remains explicit but isn't repeated in each step. Should I change the PR in that direction?

Comment (priv-kweihmann) [reply .github/workflows/pythonpackage.yml:27]: Yes, please - if we will have a knob to adjust it globally I'm fine with your suggestion

Comment (priv-kweihmann) [reply .github/workflows/pythonpackage.yml:27]: LGTM

**Pull request merged**